src/dfx-core/src/network/uri.rs

```diff
@@ -44,55 +65,19 @@ pub fn get_running_replica_port(
 ///
 /// # Prerequisites
 /// The file is expected to contain the port number only, as utf8 text.
-#[context("Failed to read port value from {}", path.to_string_lossy())]
-fn read_port_from(path: &Path) -> DfxResult<Option<u16>> {
+fn read_port_from(path: &Path) -> Result<Option<u16>, UriError> {
```

Comment:
It's true that a port is a part of a uri, but this seems like more of a NetworkConfigError